src/template.rs

```diff
@@ -141,27 +141,41 @@ pub fn walk_dir(
             continue;
         }
 
-        let filename = entry.path();
-        pbar.set_message(&filename.display().to_string());
+        let filepath = entry.path();
+        pbar.set_message(&filepath.display().to_string());
 
+        // Override the content
         let new_contents = engine
             .clone()
-            .parse_file(&filename)?
+            .parse_file(&filepath)?
             .render(&template)
             .with_context(|_e| {
                 format!(
                     "{} {} `{}`",
                     emoji::ERROR,
                     style("Error replacing placeholders").bold().red(),
-                    style(filename.display()).bold()
+                    style(filepath.display()).bold()
                 )
             })?;
-        fs::write(&filename, new_contents).with_context(|_e| {
+        fs::write(&filepath, new_contents).with_context(|_e| {
             format!(
                 "{} {} `{}`",
                 emoji::ERROR,
                 style("Error writing").bold().red(),
-                style(filename.display()).bold()
+                style(filepath.display()).bold()
+            )
+        })?;
+
+        // Check if the filename does not contains any
+        // template
+        let filename = filepath.to_str().unwrap();
+        let parsed_filename = engine.clone().parse(filename)?.render(&template)?;
+        let _ = fs::rename(&filepath, Path::new(&parsed_filename)).with_context(|_e| {
```

Comment:
what's the reason for `let _ =` here? You're already `?`-ing the `Result` away, right?

Reply:
```suggestion
        fs::rename(&filepath, Path::new(&parsed_filename)).with_context(|_e| {
```

Reply:
this smells of lingering clippy lint fixes that don't make sense anymore, so I'm going to just fix this to get this PR merged in!